aws/src/main/java/org/frankframework/filesystem/AmazonS3FileSystem.java

```diff
@@ -312,90 +338,92 @@ public OutputStream appendFile(S3Object f) {
 
 	/**
 	 * If you retrieve an S3Object, you should close this input stream as soon as possible,
-	 * because the object contents aren't buffered in memory and stream directly from Amazon S3.
-	 * Further, failure to close this stream can cause the request pool to become blocked.
+	 * because the object content is not buffered in memory and streams directly from Amazon S3.
+	 * Failure to close this stream can cause the request pool to become blocked.
 	 */
 	@Override
 	public Message readFile(S3Object file, String charset) throws FileSystemException {
 		try {
-			if(file.getObjectContent() == null) { // We have a reference but not an actual object representing the S3 bucket.
-				file = s3Client.getObject(bucketName, file.getKey()); // Fetch a new copy
+			GetObjectRequest objectRequest = GetObjectRequest.builder()
+					.bucket(bucketName)
+					.key(file.key())
+					.build();
+			ResponseInputStream<GetObjectResponse> s3ClientObject = s3Client.getObject(objectRequest);// Fetch a new copy
+
+			// Workaround for https://github.com/aws/aws-sdk-java-v2/issues/3538
+			if (s3ClientObject.response().contentLength() == 0) {
+				// Expects an empty message
+				return new Message("", FileSystemUtils.getContext(this, file, charset));
```

Comment:
```suggestion
				return Message.nullMessage(FileSystemUtils.getContext(this, file, charset));
```

Reply:
Hierdoor faalt  `org.frankframework.filesystem.FileSystemActorTest#fileSystemActorReadActionTest(org.frankframework.filesystem.FileSystemActor.FileSystemAction, boolean, boolean, java.lang.String)` dus deze change voer ik toch niet door

Reply:
Opgelost in eeb7e1d